R/basic_table_args.R

```diff
@@ -18,6 +18,11 @@
 #' @examples
 #' basic_table_args(subtitles = "SUBTITLE")
 basic_table_args <- function(...) {
+
+  if (!requireNamespace("rtables", quietly = TRUE)) {
+    stop("package rtables is required, please install")
+  }
```

Comment:
@pawelru parse_basic_table_args  uses basic_table so no need to have it in 2 places. You will get the error while using parse_basic_args because it uses basic_table and it has the error statement in there

Reply:
that's also why I didn't put the same error statement in `resolve_basic_table_args`

Reply:
@pawelru if you pass `my_custom_basic_table_args_constructor` it needs to have a specific structure that mimics `rtables::basic_table()` so I'd say you rather have rtables installed already but it's true that assertions allows NULL input and the below if statement would try to run `rtable::basic_table()`.

```{R}
  if (length(basic_table_args) == 0) {
    quote(rtables::basic_table())
  }
```

You are right. Will update the change